src/wp-includes/class-wp-list-util.php

```diff
@@ -166,9 +166,17 @@ public function pluck( $field, $index_key = null ) {
 			foreach ( $this->output as $key => $value ) {
 				if ( is_object( $value ) ) {
 					$newlist[ $key ] = $value->$field;
-				} else {
+				} elseif ( is_array( $value ) ) {
 					$newlist[ $key ] = $value[ $field ];
 				}
+
+				if ( ! ( is_object( $value ) || is_array( $value ) ) ) {
```

Comment:
```suggestion
				if ( ! is_object( $value ) && ! is_array( $value ) ) {
```
IMO, this is a little easier to read. Just a suggestion though. Same suggestion for the later instance.